lms/extractors/notebook.py

```diff
@@ -41,10 +44,29 @@ def _get_code_cells(self) -> Iterator[Cell]:
         cells = notebook['cells']
         yield from filter(self._is_code_cell, cells)
 
+    def _get_file_type(self, code_text: str) -> Tuple[str, str]:
+        if code_text:
+            first_line_end = code_text.find('\n')
+            if first_line_end == -1:
+                first_line_end = len(code_text)
+
+            first_line = code_text[:first_line_end]
+            file_type_match = self.TYPE_LINE_PREFIX.fullmatch(first_line)
+
+            if file_type_match:
+                file_type = file_type_match.group(1)
+                code_lines = code_text[first_line_end:].strip()
+            else:
+                file_type = self.DEFAULT_FILE_TYPE
```

Comment:
Define it before the if as default value, and change it only `if file_type_match`